### PR TITLE
Misc versioning tasks

### LINKED ISSUE #4203: Hash the set ID for task queue versioning data
use a base64 or hex encoded hash of the version here

_Originally posted by @dnr in https://github.com/temporalio/temporal/pull/4194#discussion_r1173418423_

### PR DESCRIPTION
**Pull request by bergundy**
- Hash build ID into set ID (Closes #4203)
- Fix persistence schema versions

### CODE REVIEW

File: schema/cassandra/temporal/versioned/v1.8/manifest.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "CurrVersion": "1.21",
```
Comment (Sushisource) [inline schema/cassandra/temporal/versioned/v1.8/manifest.json:2]: Why 1.8?

File: service/matching/version_sets.go [hunk 2]
```diff
@@ -127,6 +131,13 @@ func UpdateVersionSets(clock hlc.Clock, data *persistence.VersioningData, req *w
 	return data, nil
 }
 
+func hashBuildID(buildID string) string {
+	bytes := []byte(buildID)
+	summed := sha256.Sum256(bytes)
+	// We don't need the padding
+	return strings.ReplaceAll(base64.URLEncoding.EncodeToString(summed[:]), "=", "")
```
Comment (Sushisource) [inline service/matching/version_sets.go:138]: We mentioned that we might not need all the characters, but that's not a biggie.

**Review by Sushisource — APPROVED** [review]

File: service/matching/version_sets.go (hunk 2 above)
Comment (dnr) [inline service/matching/version_sets.go:138]: first, `RawURLEncoding` doesn't include padding.
second, just truncate instead:
```suggestion
	return base64.RawURLEncoding.EncodeToString(summed[:])[:20]
```
(20 base64 chars is 120 bits of entropy which is plenty)

File: service/matching/version_sets_test.go [hunk 3]
```diff
@@ -41,7 +41,7 @@ import (
 
 func mkNewSet(id string, clock clockpb.HybridLogicalClock) *persistencepb.CompatibleVersionSet {
 	return &persistencepb.CompatibleVersionSet{
-		SetIds:                 []string{id},
+		SetIds:                 []string{hashBuildID(id)},
```
Comment (dnr) [inline service/matching/version_sets_test.go:44]: no need to hash in the tests if it makes them more readable. but also fine to do it for consistency.

File: schema/cassandra/temporal/versioned/v1.8/manifest.json (hunk 1 above)
Comment (dnr) [reply schema/cassandra/temporal/versioned/v1.8/manifest.json:2]: presumably it's the next sequential version?

Comment (bergundy) [reply schema/cassandra/temporal/versioned/v1.8/manifest.json:2]: Yes, it's the next sequential version, I wrongly assumed this version is tied to a server release initial, this change fixes that.

File: service/matching/version_sets.go (hunk 2 above)
Comment (bergundy) [reply service/matching/version_sets.go:138]: Sure

File: service/matching/version_sets_test.go (hunk 3 above)
Comment (bergundy) [reply service/matching/version_sets_test.go:44]: It just simplifies the the test logic because the implementation hashes the build IDs.

**Review by dnr — APPROVED** [review]

**Pull request merged**